test/e2e/apm/association_test.go

```diff
@@ -85,6 +85,92 @@ func TestAPMKibanaAssociation(t *testing.T) {
+func TestAPMKibanaAssociationBasePathConfiguredInConfig(t *testing.T) {
+	stackVersion := version.MustParse(test.Ctx().ElasticStackVersion)
+	if !stackVersion.GTE(apmv1.ApmAgentConfigurationMinVersion) {
+		t.SkipNow()
+	}
+
+	ns := test.Ctx().ManagedNamespace(0)
+	name := "test-apm-kb-assoc"
+
+	esBuilder := elasticsearch.NewBuilder(name).
+		WithNamespace(ns).
+		WithESMasterDataNodes(2, elasticsearch.DefaultResources). // TODO: revert when https://github.com/elastic/cloud-on-k8s/issues/7418 is resolved.
+		WithRestrictedSecurityContext()
+
+	kbBuilder := kibana.NewBuilder(name).
+		WithNamespace(ns).
+		WithElasticsearchRef(esBuilder.Ref()).
+		WithNodeCount(1).
+		WithRestrictedSecurityContext().
+		WithAPMIntegration().
+		WithConfig(map[string]interface{}{
+			"server.basePath":        "/monitoring/kibana",
+			"server.rewriteBasePath": true,
+		})
+
+	apmBuilder := apmserver.NewBuilder(name).
+		WithNamespace(ns).
+		WithElasticsearchRef(esBuilder.Ref()).
+		WithKibanaRef(kbBuilder.Ref()).
+		WithNodeCount(1).
+		WithRestrictedSecurityContext().
+		WithConfig(map[string]interface{}{
+			"apm-server.ilm.enabled":                           false,
+			"setup.template.settings.index.number_of_replicas": 0, // avoid ES yellow state on a 1 node ES cluster
+		})
+
+	test.Sequence(nil, test.EmptySteps, esBuilder, kbBuilder, apmBuilder).RunSequential(t)
+}
+
+// TestAPMKibanaAssociationBasePathConfiguredInENV tests associating an APM Server with Kibana with a custom basePath configured as an ENV.
+func TestAPMKibanaAssociationBasePathConfiguredInENV(t *testing.T) {
+	stackVersion := version.MustParse(test.Ctx().ElasticStackVersion)
+	if !stackVersion.GTE(apmv1.ApmAgentConfigurationMinVersion) {
+		t.SkipNow()
+	}
+
+	ns := test.Ctx().ManagedNamespace(0)
+	name := "test-apm-kb-assoc"
+
+	esBuilder := elasticsearch.NewBuilder(name).
+		WithNamespace(ns).
+		WithESMasterDataNodes(2, elasticsearch.DefaultResources). // TODO: revert when https://github.com/elastic/cloud-on-k8s/issues/7418 is resolved.
+		WithRestrictedSecurityContext()
+
+	kbBuilder := kibana.NewBuilder(name).
+		WithNamespace(ns).
+		WithElasticsearchRef(esBuilder.Ref()).
+		WithNodeCount(1).
+		WithRestrictedSecurityContext().
+		WithAPMIntegration().
+		WithENV([]corev1.EnvVar{
+			{
+				Name:  "SERVER_BASEPATH",
+				Value: "/monitoring/kibana",
+			},
+			{
+				Name:  "SERVER_REWRITEBASEPATH",
+				Value: "true",
+			},
+		})
+
+	apmBuilder := apmserver.NewBuilder(name).
+		WithNamespace(ns).
+		WithElasticsearchRef(esBuilder.Ref()).
+		WithKibanaRef(kbBuilder.Ref()).
+		WithNodeCount(1).
+		WithRestrictedSecurityContext().
+		WithConfig(map[string]interface{}{
+			"apm-server.ilm.enabled":                           false,
+			"setup.template.settings.index.number_of_replicas": 0, // avoid ES yellow state on a 1 node ES cluster
+		})
+
+	test.Sequence(nil, test.EmptySteps, esBuilder, kbBuilder, apmBuilder).RunSequential(t)
+}
```

Comment:
This is fine. I think we shoudl try to keep e2e execution times low though. I wonder if it would be quicker to use https://github.com/elastic/cloud-on-k8s/blob/e74296fa4cb39f58280d25c8f0bde6ed1404b0c1/test/e2e/test/run_mutation.go#L14

So basically have just one test and change the Kibana to use env vars, which might be faster than recreating the whole stack twice. Something along the lines of (untested!): 

```golang
	kbWithEnv := kbBuilder.DeepCopy().WithEnv([]corev1.EnvVar{
		{
			Name:  "SERVER_BASEPATH",
			Value: "/monitoring/kibana",
		},
		{
			Name:  "SERVER_REWRITEBASEPATH",
			Value: "true",
		},
	}).WithConfig(nil)

	test.RunMutations(t, []test.Builder{esBuilder, kbBuilder, apmBuilder}, []test.Builder{kbWithEnv})
```

Reply:
Thanks, I was not aware of this. I have made this change, this makes it much quicker and simpler